R/as_gt.R

```diff
@@ -305,21 +305,32 @@ as_gt.gs_design <- function(
     footnote = NULL,
     display_bound = c("Efficacy", "Futility"),
     display_columns = NULL,
-    display_inf_bound = TRUE,
+    display_inf_bound = FALSE,
     full_alpha = 0.025,
     ...) {
+
   method <- class(x)[class(x) %in% c("ahr", "wlr", "combo", "rd")]
+
   x_alpha <- max((x %>% dplyr::filter(Bound == display_bound[1]))[[colname_spannersub[2]]])
+
   x_non_binding <- "non_binding" %in% class(x)
+
   x_k <- lapply(x$Analysis, function(x) {
     return(as.numeric(substring(x, 11, 11)))
   }) %>% unlist()
+
+  if (!display_inf_bound) {
+    x <- x %>% filter(!is.infinite(Z))
+  }
+
   x_old <- x
 
   # Set defaults ----
   # set different default title to different methods
   if (method == "ahr" && is.null(title)) {
-    title <- "Bound summary for AHR design"
+    title <- ifelse("updated_design" %in% class(x),
+                    "Updated Bound summary for AHR design",
+                    "Bound summary for AHR design")
```

Comment:
The canonical style is:

```r
title <- if (inherits(x, "updated_design")) "Updated Bound summary for AHR design" else "Bound summary for AHR design"
```

See https://style.tidyverse.org/syntax.html?q=ifelse#if-statements.